src/budget_transaction.rs

```diff
@@ -83,7 +83,7 @@ impl BudgetTransaction for Transaction {
         let budget = Budget::Pay(payment);
         let instruction = Instruction::NewContract(Contract { budget, tokens });
         let userdata = serialize(&instruction).unwrap();
-        Self::new(
+        Self::new_one_program(
```

Comment:
Looks like `new` was the right name. What happened here?

Reply:
@garious i am ok with either one.  Just seemed that `new` was the most general interface.  I can name this one new and the other one `new_multi_instruction`.  How does that sound?

Reply:
Yes, sounds good.  Use `new` for the most common, not the most general.  The most general one will keep changing as we add parameters. That'd be a big pain to keep updating all the earlier, probably simpler, use cases.